src/core/load-fragment.c

```diff
@@ -5898,6 +5899,14 @@ int config_parse_bpf_foreign_program(
         if (r < 0)
                 return 0;
 
+        r = path_is_fs_type(resolved, BPF_FS_MAGIC);
+        if (r < 0)
+                return log_error_errno(r, "Failed to determine filesystem type of %s: %m", resolved);
+        if (r == 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, SYNTHETIC_ERRNO(EINVAL), "Path in BPF filesystem is expected, ignoring: %s", rvalue);
+                return 0;
+        }
+
```

Comment:
When the unit file is loaded, the path may not be mounted? In stead of here, the fs type should be checked when loading the BPF program?

Reply:
Hmm, indeed. Doing this when parsing config is too early.

Reply:
Nice catch, thanks. Moved to `cgroup_context_apply`.